In [ ]:
# COLAB에서 작동할 때 사용할 것

#from google.colab import drive
#drive.mount('/content/drive')
# %cd /content/drive/MyDrive/AIVLE/데이터 분석/3. 데이터 전처리

In [1]:
# GitHub와 연동된 Local 환경일 때 사용
%cd ./

c:\Users\Hanasaki\바탕 화면\AIVLE\데이터 분석\3. 데이터 전처리


---
<br/>

<br/>
<br/>

## 먼저  pandas 라이브러리를 불러옵니다.



In [2]:
import pandas as pd



# 데이터 불러오기: read_csv()

이제 data폴더에 저장되어있는 데이터인 **seoul_park.csv** 파일을 불러와 데이터프레임 `df`에 저장합니다. csv 파일 형태를 가지므로 `read_csv()`를 활용하여 불러옵니다. `read_csv()`에는 데이터가 저장되어있는 파일의 경로를 입력받습니다.

* read_csv()
* read_excel()
* read_json()

In [3]:
# 여기에 코드를 작성해보세요!
dframe = pd.read_csv("./data/seoul_park.csv", encoding='utf-8')
dframe

날짜 공휴일     날씨   유료합계     어른  청소년  어린이  외국인   단체   무료합계     총계
0     2016-01-01   O  구름 조금  3,359  2,799  141  419   47    0  1,023  4,382
1     2016-01-02   O  구름 많음  5,173  4,370  203  600  100  111  2,092  7,265
2     2016-01-03   O  구름 많음  3,008  2,571  128  309   91    0  1,549  4,557
3     2016-01-04   X  구름 많음    890    602  NaN  235   51  223    800  1,690
4     2016-01-05   X  구름 많음    416    319   35   62   43   47    840  1,256
...          ...  ..    ...    ...    ...  ...  ...  ...  ...    ...    ...
1081  2019-03-27   X  구름 많음    504    464   10   30   21    -    613  1,117
1082  2019-03-28   X  구름 많음    761    687   46   28   35  108    904  1,665
1083  2019-03-29   X  구름 조금  1,644  1,447  120   77   14  188  1,226  2,870
1084  2019-03-30   O     흐림  1,539  1,326   44  169   29  115    913  2,452
1085  2019-03-31   O  구름 조금  3,061  2,563  111  387   53    -  1,357  4,418

[1086 rows x 11 columns]

In [4]:
# 모범답안

# 파일 불러오기
df = pd.read_csv("./data/seoul_park.csv")
df

날짜 공휴일     날씨   유료합계     어른  청소년  어린이  외국인   단체   무료합계     총계
0     2016-01-01   O  구름 조금  3,359  2,799  141  419   47    0  1,023  4,382
1     2016-01-02   O  구름 많음  5,173  4,370  203  600  100  111  2,092  7,265
2     2016-01-03   O  구름 많음  3,008  2,571  128  309   91    0  1,549  4,557
3     2016-01-04   X  구름 많음    890    602  NaN  235   51  223    800  1,690
4     2016-01-05   X  구름 많음    416    319   35   62   43   47    840  1,256
...          ...  ..    ...    ...    ...  ...  ...  ...  ...    ...    ...
1081  2019-03-27   X  구름 많음    504    464   10   30   21    -    613  1,117
1082  2019-03-28   X  구름 많음    761    687   46   28   35  108    904  1,665
1083  2019-03-29   X  구름 조금  1,644  1,447  120   77   14  188  1,226  2,870
1084  2019-03-30   O     흐림  1,539  1,326   44  169   29  115    913  2,452
1085  2019-03-31   O  구름 조금  3,061  2,563  111  387   53    -  1,357  4,418

[1086 rows x 11 columns]

<br/>
<br/>

## 데이터 저장하기 : to_csv()

작업중인 DataFrame 을 csv파일로 저장하려면 to_csv()를 활용합니다.
원하는 파일 형식에 따라 to_json(), to_excel() 로 활용할 수 있습니다.

In [ ]:
# 여기에 코드를 작성해보세요!
dframe.to_csv("./data/testme.csv")

In [ ]:
# 모범답안

# 파일 저장하기 -> 이후 파일이 생성되었는지 확인하세요

df.to_csv("./data/test.csv")

<br/>
<br/>

## 데이터 살펴보기: shape
DataFrame 에서 행의 갯수, 열의 갯수는 shape 메서드를 통해 확인할 수 있습니다.

In [ ]:
# 여기에 코드를 작성해보세요!
dframe.shape

In [ ]:
# 모범답안

df.shape  # (행 , 열)

<br/>
<br/>

## 데이터 살펴보기: head(), tail()
데이터의 갯수는 총 1086개로 데이터가 너무 커 한눈에 들어오지 않습니다. pandas 라이브러리에는 데이터를 살펴보기에 유용한 기능들이 있습니다.
`head()`, `tail()`을 사용하면 데이터의 처음과 끝의 일부를 확인할 수 있습니다.

In [ ]:
# 여기에 코드를 작성해보세요!

In [ ]:
# 모범답안

df.head() # 기본적으로 5개

In [ ]:
# 여기에 코드를 작성해보세요!

In [ ]:
# 모범답안

df.tail()

`head()`나 `tail()`메서드 안에 숫자를 입력하여 출력하는 데이터의 갯수를 지정할 수도 있습니다.

In [ ]:
# 여기에 코드를 작성해보세요!

In [ ]:
# 모범답안

df.head(10)

## 데이터 정보 확인: info()

데이터가 몇 개의 값을 가지는지, 어떤 자료형으로 저장되어있는지를 확인하는데에는 `info()`를 활용할 수 있습니다.

In [ ]:
# 여기에 코드를 작성해보세요!

In [ ]:
# 모범답안

df.info()

## 데이터 숫자 세기: value_counts()
`value_counts()`를 활용하면 컬럼에 어떤 값들이 몇 개씩 존재하는지 확인할 수 있습니다.

예를 들어 `df`의 날씨 컬럼에는 날씨가 저장되어 있는데 해당 컬럼에 `value_counts()`를 활용하면 어떤 종류의 날씨들이 있는지, 각 날씨별로 몇 개의 데이터가 있는지 확인할 수 있습니다.

In [ ]:
# 여기에 코드를 작성해보세요!

In [ ]:
# 모범답안

df["날씨"].value_counts()

<br/>
<br/>

## 데이터 통계정보 보기 : describe()
DataFrame에서 숫자형 변수의 기초 통계량을 확인할 수 있습니다.
describe() 메서드를 사용하면 평균, 표준편차, 최대값, 최소값, 사분위수 등의 요약 통계 정보를 한눈에 볼 수 있습니다.

In [ ]:
# 여기에 코드를 작성해보세요!

In [ ]:
# 모범답안

df.describe()